### ISSUE TITLE
Add get_lr in ReduceLROnPlateau

### ISSUE THREAD

**Issue by mlix11** [pytorch/pytorch#2829]
With Learning-Rate Scheduler a nice enhancement came with the update, beside more others.

But why does the ReduceLROnPlateau not have a **get_lr method**, for consistency it should have one? Because every other Scheduler have one and omit a **verbose-mode**.

**Comment by Jiaming-Liu** [comment]
Hi, I'm the puller of lr_scheduler.

Actually, `get_lr` is designed to be more like an internal function for computing instead of fetching the learning rate. Sorry for the confusion.

For fetching, you could use `[ group['lr'] for group in optim.param_groups ]` for now (Yep, it's tedious). Maybe what needs to be done is to rename get_lr to compute_lr and create another function called get_lr.

**Comment by loose11** [comment]
But fetching is very interesting, if you want to analyze the learning-rate and your loss for example. Is there a plan to clean this up?

**Comment by apaszke** [comment]
The problem with having a function like this, is that a scheduler can maintain LR for multiple groups, with different values, so there's no single good answer

**Comment by mruberry** [comment]
Closing due to age.

**Comment by sweihub** [comment]
What a bad design, the Schedulers dose not have a consistent method, and the issue was closed due to no solution.

**Comment by H4dr1en** [comment]
Landed here because of https://github.com/pytorch/ignite/issues/462

I ended up patching ReduceLROnPlateau as follows:

````
class ReduceLROnPlateauPatch(ReduceLROnPlateau, _LRScheduler):
    def get_lr(self):
        lrs = [ group['lr'] for group in self.optimizer.param_groups ]
        if not all (lr == lrs[0] for lr in lrs):
            raise ValueError("Learning rate of all optimizer.param_groups should be the same.")
        return lrs[0]
```

Most likely assuming all learning rates are the same is a good basis for most of the users (I leave it to you), this hack could be added to pytorch. That would also give to users a better understanding of how/where to adapt the code for their use case. WDYT?

**Comment by JurijsNazarovs** [comment]
> Still hope 'ReduceLROnPlateau' can have attribute 'get_lr'

You can use `._last_lr` instead. So, `print("lr:", lr_scheduler._last_lr)`

**Comment by pbelcak** [comment]
> The problem with having a function like this, is that a scheduler can maintain LR for multiple groups, with different values, so there's no single good answer

This appears to no longer be a valid explanation. `.get_last_lr()` now returns a list of learning rates that is indexed per param group.
